src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -648,6 +650,89 @@ class ActorIdStringGenerator : public Generator<std::string> {
     std::string _actorId;
 };
 
+/** `{^Now: {}}` */
+class NowGenerator : public Generator<bsoncxx::types::b_date> {
+public:
+    NowGenerator(const Node& node, GeneratorArgs generatorArgs) {}
+    bsoncxx::types::b_date evaluate() override {
+        return bsoncxx::types::b_date{std::chrono::system_clock::now()};
```

Comment:
hrm. this starts to represent a departure from existing behavior where we run the same workload twice and it results in the same sequence of values being generated. That's probably exactly what the user expects in this case, but it makes me a bit hesitant. Maybe we should start the clock at some arbitrary time-point at the beginning of the workload and then make now() be an offset? Just a thought.

Reply:
I'm not sure how much "real" time values matters. I was imagining "now" could start as `2020-12-18T00:00:00.00000` or something and then every subsequent call to `now` could increment the current clock by a random amount or something. Or perhaps it could increment the fake clock by the amount of time that has elapsed since the workload has started.

I don't know the full use-case for wanting `now()` though. We don't really need to worry about it being the same across clusters because the workload-client is the only machine that would see the value unless queries use `$currentDate` and friends.

(I'm fine with however you decide to proceed here, imho, as long as we have clear messaging _somewhere_ about the implications of repro-ability for workloads that use `^Now`.)

Reply:
I'm going to mull on this a bit more. I think we should do as built for now, but see if we can do something better. I worry about some interaction with the running system and real times coming up, but don't have a handy example so maybe what you suggest would work. If we did something like this I would like to fix the test start time, and then advance from their proportional to real time. That is, if the server got faster, there could end up more documents within a specific datetime range.